### ISSUE TITLE
Nodeos 4.0.1 segfault on startup

### ISSUE THREAD

**Issue by matthewdarwin** [AntelopeIO/leap#1260]
After nodeos was cleanly stopped to take a backup, it segfaults on startup.

```
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.163 nodeos    net_api_plugin.cpp:53         plugin_startup       ] starting net_api_plugin
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.163 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/connect
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.163 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/connections
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.163 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/disconnect
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.163 nodeos    http_plugin.cpp:378           add_async_handler    ] add api url: /v1/net/status
Jun  9 05:47:05 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:05.164 nodeos    producer_plugin.cpp:1227      plugin_startup       ] producer plugin:  plugin_startup() begin
Jun  9 05:47:07 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:07.850 nodeos    producer_plugin.cpp:1321      plugin_shutdown      ] exit shutdown
Jun  9 05:47:07 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:07.857 nodeos    net_plugin.cpp:3896           plugin_shutdown      ] shutdown..
Jun  9 05:47:07 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:07.857 nodeos    net_plugin.cpp:2722           plugin_shutdown      ] close 0 connections
Jun  9 05:47:07 telostest-ship11 nodeos[1736269]: info  2023-06-09T05:47:07.865 nodeos    net_plugin.cpp:3900           plugin_shutdown      ] exit shutdown
```

**Comment by matthewdarwin** [comment]
Matt W notices the timing is > 1 second

```
info  2023-06-09T05:47:05.164 nodeos    producer_plugin.cpp:1227      plugin_startup       ] producer plugin:  plugin_startup() begin
info  2023-06-09T05:47:07.850 nodeos    producer_plugin.cpp:1321      plugin_shutdown      ] exit shutdown
```

**Comment by spoonincode** [comment]
Did you have read only trx threads enabled on this node?

**Comment by heifner** [comment]
Since `chain_api_plugin` is enabled then `read-only-threads` defaults to 3. If this is not a public API node (assuming it is not because it is SHiP), then you should run with `read-only-threads = 0`

**Comment by matthewdarwin** [comment]
Thanks for the hint.  chain_api_plugin is enabled because we need `/v1/chain/get_info`.

**Comment by spoonincode** [comment]
okay, then I'm going to guess related to #1223 and fixed by #1221

**Comment by matthewdarwin** [comment]
@stephenpdeos would be good to get a 4.0.2 relaese. ;-)
